### DISCUSSION TITLE
Multiple TraceId on same thread

### DISCUSSION THREAD

**Discussion by schelwa-tibco** [open-telemetry/opentelemetry-java#2997] [Q&A]
I am seeing multiple traceId in one job execution. Is it possible that multiple traceIds are created for one job execution? My next question, what api call decides to end a trace or start a new trace, especially when I am queueing up multiple jobs to execute on same thread?

**Comment by jkwatson** [comment]
If you want to start a new trace, you want to explicitly `setNoParent()` on the span builder.

It certainly possible that you get multiple trace ids from one "job execution" depending on the instrumentation. Can you give some details about what instrumentation you're using? Are you using the javaagent, or doing all manual instrumentation?

**Reply by schelwa-tibco** [reply]
I am using manual instrumentation with autoconfigure extension and kept everything default. In every method that I am instrumenting, I create a span at the start of method and end it in finally. Use case is simple. I am not sure what could cause a end of a trace.

**Reply by jkwatson** [reply]
Can you paste some code snippets that show how you're writing the instrumentation? Are you putting spans into the Context via `Span.makeCurrent()` so that they get propagated as the parent span (and then making sure the resulting Scope gets closed using try-with-resources)?

**Reply by schelwa-tibco** [reply]
I am using `context.with(span)` followed by `context.makeCurrent()`, is that same as `Span.makeCurrent()`? 

In my case, in one request execution(executed on a single thread), first 5 instrumented method have spans with same traceId except the last instrumented method has span with different traceId.

Here is a code snippet, I used in all the instrumented methods. 

```
Context currentContext = Context.current();
Span span = tracer.spanBuilder(spanName).setParent(currentContext).setParent(SpanKind.SERVER).startSpan();
Context context = currentContext.with(span);
scope = context.makeCurrent();

// request processing code here

// finally
span.end();
scope.close();
```

**Reply by jkwatson** [reply]
It's hard to say whether you're doing things correctly or not, but here's a gist that has a functioning example of how to properly manage propagation and scope: https://gist.github.com/jkwatson/323feb711971e5f47ea3b9379d94f954
